Modules/IO/ImageBase/test/itkWriteImageFunctionGTest.cxx

```diff
@@ -58,22 +59,30 @@ struct ITKWriteImageFunctionTest : public ::testing::Test
 
 TEST_F(ITKWriteImageFunctionTest, ImageTypes)
 {
+  const std::string fileName = "test1.mha";
+
   ImageType::Pointer image_ptr = MakeImage();
-  itk::WriteImage(image_ptr, "test1.mha");
-  itk::WriteImage(std::move(image_ptr), "test1.mha");
+  itk::WriteImage(image_ptr, fileName);
+  itk::WriteImage(std::move(image_ptr), fileName);
 
   ImageType::ConstPointer image_cptr = MakeImage();
-  itk::WriteImage(image_cptr, "test1.mha");
-  itk::WriteImage(std::move(image_cptr), "test1.mha");
+  itk::WriteImage(image_cptr, fileName);
+  itk::WriteImage(std::move(image_cptr), fileName);
 
   const ImageType::ConstPointer image_ccptr = MakeImage();
-  itk::WriteImage(image_ccptr, "test1.mha");
-  itk::WriteImage(std::move(image_ccptr), "test1.mha");
+  itk::WriteImage(image_ccptr, fileName);
+  itk::WriteImage(std::move(image_ccptr), fileName);
 
   image_ptr = MakeImage();
   ImageType * image_rptr = image_ptr.GetPointer();
-  itk::WriteImage(image_rptr, "test1.mha");
+  itk::WriteImage(image_rptr, fileName);
 
   const ImageType * image_crptr = image_ptr.GetPointer();
-  itk::WriteImage(image_crptr, "test1.mha");
+  itk::WriteImage(image_crptr, fileName);
+
+  // Test that what is written is also read back.
```

Comment:
Maybe this should be done after each WriteImage? Also maybe not, as we don't want to waste time needlessly.

Reply:
Good question @dzenanz but I'm not sure either, as I feel that the added value of such extra tests seems rather small, while there could indeed be a waste of time. Maybe someone else who _really_ wants to have a test after each `WriteImage` could submit a follow-up PR  😸